frontend/src/scenes/ingestion/panels/VerificationPanel.tsx

```diff
@@ -7,26 +7,35 @@ import { teamLogic } from 'scenes/teamLogic'
 import { Spinner } from 'lib/components/Spinner/Spinner'
 import { LemonButton } from 'lib/components/LemonButton'
 import './Panels.scss'
-import { IconCheckCircleOutline } from 'lib/components/icons'
 import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
 import { EventBufferNotice } from 'scenes/events/EventBufferNotice'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { FEATURE_FLAGS } from 'lib/constants'
 
 export function VerificationPanel(): JSX.Element {
     const { loadCurrentTeam } = useActions(teamLogic)
     const { currentTeam } = useValues(teamLogic)
-    const { setVerify, completeOnboarding } = useActions(ingestionLogic)
-    const { index } = useValues(ingestionLogic)
+    const { setAddBilling, completeOnboarding, setCurrentIndex } = useActions(ingestionLogic)
     const { reportIngestionContinueWithoutVerifying } = useActions(eventUsageLogic)
+    const { featureFlags } = useValues(featureFlagLogic)
+
+    const shouldShowBilling = featureFlags[FEATURE_FLAGS.ONBOARDING_BILLING]
 
     useInterval(() => {
         if (!currentTeam?.ingested_event) {
             loadCurrentTeam()
         }
     }, 2000)
 
+    useEffect(() => {
+        if (!shouldShowBilling) {
+            setCurrentIndex(3)
+        }
+    }, [currentTeam?.ingested_event])
```

Comment:
You can also do this with a [subscription](https://keajs.org/docs/plugins/subscriptions) in the logic.